### DISCUSSION TITLE
Changing the fork

### DISCUSSION THREAD

**Discussion by prashant1804** [LazyVim/LazyVim#882] [General]
How can I change the fork of a particular plugin. When I try to do so, it gives me error that their are two. Is there such a ways to change it directly from starter config rather than changing it at base level.

**Comment by folke** [comment]
check the lazy.nvim docs for the `dev` option

**Reply by l1n3n01z** [reply]
Hi, the dev option works well for plugins, but I think it is not documented how to set the dev options to load plugins from a specific directory. I mean the dev table that lazy.nvim uses to configure local plugin directories and patterns and such.

I tried several things, but ended up creating a soft link from ~/projects to the directory I want to keep my local plugins in.

And in general, how do you set the lazy.nvim options from LazyVim?

**Reply by dpetka2001** [reply]
#1092 and #1129 might be of help to you.

**Comment by l1n3n01z** [comment]
> #1092 and #1129 might be of help to you.

So, please bear with me. I'm not trying to use a fork of LazyVim as currently I want to keep LazyVim as vanilla as I can. So it doesn't seem like the solutions to these issues apply to me.

I have a fork of a specific plugin in a specific folder. On setup of lazy.nvim, it's possible to give the package manager some options, i.e. dev = {path="/path/to/folder_full_of_local_nvim_plugins"} which changes the default path where it looks for dev plugins from '~/projects'.

However in LazyVim I don't understand how to set this option and it also seems to be the only possible way to set the path of where plugins reside. I thought maybe you could set it per plugin, but I have not been able to find any way to do that either.

**Comment by l1n3n01z** [comment]
> ... I thought maybe you could set it per plugin, but I have not been able to find any way to do that either.

Turns out I can actually set the "dir" option per plugin (I just overlooked that in the documentation for the plugin spec, saw it when reading through lazy.nvim source code lol), and that actually seems to work, but I would prefer to set lazy.nvim dev path and just set dev=true per plugin. 

Which works for vanilla lazy.nvim but not been able to get it working in LazyVim.

**Reply by dpetka2001** [reply]
I just have the following in `~/.config/nvim/lua/config/lazy.lua` inside the the `lazy.nvim` setup
```lua
  dev = {
    path = "~/projects/plugins",
    patterns = { "LazyVim", "lazyflex" },
  },
```
and have the local forked repos under `~/projects/plugins` in my case. You can change the `path` to whatever you want.

**Comment by l1n3n01z** [comment]
> I just have the following in `~/.config/nvim/lua/config/lazy.lua` inside the the `lazy.nvim` setup
> 
> ```lua
>   dev = {
>     path = "~/projects/plugins",
>     patterns = { "LazyVim", "lazyflex" },
>   },
> ```
> 
> and have the local forked repos under `~/projects/plugins` in my case. You can change the `path` to whatever you want.

This is exactly what I was looking for. And I was so sure I tried it, but the lua lsp must have convinced me that it wasn't possible, as it claims that dev is only a boolean. 

This worked for me
```lua
require("lazy").setup({
  -- ... some other options
  spec = {
    -- .. some stuff
  },
  dev = {
    path = "~/personal/nvim-plugins/",
  },
  -- ... some other options
})
```

**Reply by dpetka2001** [reply]
Yes, that `dev` must be boolean is some kind of misconception from `lua_ls`, because it can be used both as `dev = true` inside a plugin spec and as a table in `lazy.lua` to specify where the `dev.path` is located. Nothing to worry about.
